### PR TITLE
fix logic clear status and save status

### PR DESCRIPTION
**Pull request by dukenv0307**
### Fixed Issues

$ https://github.com/Expensify/App/issues/33735
PROPOSAL: https://github.com/Expensify/App/issues/33735#issuecomment-1871662505

### Tests

1. Log in with any account.
2. Open Settings/ Profile/ Status
3. Verify that it shows a disabled emoji icon and select a any emoji
4. Verify that status emoji appears
5. Clear status and verify that status emoji cleared
6. Press Save button
7. Cleared status is not visible if press save button

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Log in with any account.
2. Open Settings/ Profile/ Status
3. Verify that it shows a disabled emoji icon and select a any emoji
4. Verify that status emoji appears
5. Clear status and verify that status emoji cleared
6. Press Save button
7. Cleared status is not visible if press save button

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/129500732/47a40333-718a-47bb-8c1c-d51ad9abe298

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/129500732/de078512-7863-4848-8d54-0fafdc5030c5

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/129500732/2332f219-c837-4ae0-997d-68fe184f44ca

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/129500732/ca28a15b-034b-4589-9c4c-7b3e7b25c25b

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
> One thing I think it a little incongruent is that if the user has the placeholder emoji selected and the status is empty it will set the default speech bubble emoji

My understanding is that this is a known bug and is being fixed? cc @amyevans for thoughts there

**Comment by amyevans** [comment]
> My understanding is that this is a known bug and is being fixed? cc @amyevans for thoughts there

🤔 Do you know _where_ it is being fixed? IMO once the message becomes empty, we should remove the default speech bubble emoji and display the placeholder emoji (And handle that in this PR if we don't have a distinct GH for it elsewhere.)

**Comment by shawnborton** [comment]
Ah, I thought you commented in [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1705933176047769?thread_ts=1705893084.906069&cid=C049HHMV9SM) that it would be fixed here: https://github.com/Expensify/App/pull/34780

But now I am realizing... here is this PR 🤣 

So yeah, let's fix that behavior here.

**Comment by amyevans** [comment]
Lol okay I thought that might've been the source of the confusion but wasn't sure 😄

**Comment by shawnborton** [comment]
That sounds correct to me.

**Comment by shawnborton** [comment]
Although, I wonder if we should allow a user to save a status that is just the emoji and no message?

If they save a message that doesn't have an emoji, perhaps we just default to a comment bubble or we throw a form error?

**Comment by amyevans** [comment]
> Although, I wonder if we should allow a user to save a status that is just the emoji and no message?

Yeah I think we should definitely allow for this.

> If they save a message that doesn't have an emoji, perhaps we just default to a comment bubble or we throw a form error?

I think defaulting to a comment bubble is good (which is what we're already doing)

So the allowable options would be:
* custom emoji + message
* custom emoji + no message
* message only -> comment bubble emoji forced
* placeholder emoji + no message = status cleared

**Comment by shawnborton** [comment]
Cool cool, works for me!

**Comment by dannymcclain** [comment]
Yeah that sounds good to me too 🚀

**Comment by amyevans** [comment]
> Did you mean that if the user deletes their status (by pressing backspace for example) when the text input is empty the status emoji should revert to the placeholder?

Yes, that's what I was envisioning. However I'll let @Expensify/design weigh in again too!

**Comment by dannymcclain** [comment]
I don't have super strong feelings about that—could honestly go either way. I am liking what I'm seeing in [this video](https://github.com/Expensify/App/pull/34780#issuecomment-1907302425) though! It's comin' along!

**Comment by shawnborton** [comment]
Hmm not entirely sure what my opinion is there. I think if there happens to be an emoji present in the emoji selector, the user should have to press clear status or pick a different emoji. I don't think I would expect that clearing the text input would change the emoji, mostly because we are supporting that users could have an emoji-only status right?

**Comment by dannymcclain** [comment]
> I think if there happens to be an emoji present in the emoji selector, the user should have to press clear status or pick a different emoji. I don't think I would expect that clearing the text input would change the emoji

You know what, now that you lay it out like this, I'm more in favor of making the user press clear or select a different emoji. So NOT resetting the emoji if they delete the message. That's my vote.

**Comment by amyevans** [comment]
Right, yes, I should have clarified I would expect it to be cleared _only if_ it was the comment bubble emoji.

**Comment by dannymcclain** [comment]
> Right, yes, I should have clarified I would expect it to be cleared only if it was the comment bubble emoji.

Even so, I think if you want to clear it you should click the explicit `Clear status` button. `:my2cents:`

**Comment by amyevans** [comment]
That's fair, and I definitely don't feel passionately about it. I'm happy to move forward with the functionality as-is in the PR!

**Review by Ollyws — APPROVED** [review]
LGTM.

**Review by amyevans — APPROVED** [review]

File: src/components/EmojiPicker/EmojiPickerButtonDropdown.js [hunk 1]
```diff
@@ -60,7 +60,12 @@ function EmojiPickerButtonDropdown(props) {
                             style={styles.emojiPickerButtonDropdownIcon}
                             numberOfLines={1}
                         >
-                            {props.value}
+                            {props.value || (
+                                <Icon
+                                    src={Expensicons.Emoji}
+                                    fill={StyleUtils.getIconFillColor(CONST.BUTTON_STATES.DISABLED)}
+                                />
+                            )}
```
Comment (situchan) [inline src/components/EmojiPicker/EmojiPickerButtonDropdown.js:68]: Putting `Icon` as child of `Text` caused [regression](https://github.com/Expensify/App/issues/80427#issuecomment-4261379818).
We should avoid Non-text view elements inside Text as this is not working correctly on native.
More details: https://github.com/Expensify/App/issues/80427#issuecomment-3808052192

**Pull request merged**